File: src/models/Item.ts
export class Item {
  private rented: boolean = false;
  private rentalStartDate: string | null = null;
  private rentalEndDate: string | null = null;

  constructor(
    public readonly id: number,
    public readonly name: string
  ) {}

  public isAvailable(): boolean {
    return !this.rented;
  }

  public rent(startDate: string, endDate: string): boolean {
    if (!this.isAvailable()) {
      return false;
    }

    if (this.isValidRentalPeriod(startDate, endDate)) {
      this.rented = true;
      this.rentalStartDate = startDate;
      this.rentalEndDate = endDate;
      return true;
    }
    
    return false;
  }

  private isValidRentalPeriod(startDate: string, endDate: string): boolean {
    const start = new Date(startDate);
    const end = new Date(endDate);
    const minRentalDuration = 24 * 60 * 60 * 1000;
    return (end.getTime() - start.getTime()) >= minRentalDuration;
  }

  public getRentalDetails(): string {
    if (this.rented) {
      return `Rented from ${this.rentalStartDate} to ${this.rentalEndDate}`;
    }
    return "Available";
  }
}

File: src/main/java/com/mycompany/location/location/model/Item.java
package com.location.models;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class Item {
    private final int id;
    private final String name;
    private boolean isRented = false;
    private LocalDate rentalStartDate;
    private LocalDate rentalEndDate;

    public Item(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public boolean isAvailable() {
        return !isRented;
    }

    public boolean rent(LocalDate startDate, LocalDate endDate) {
        if (isAvailable() && !startDate.isAfter(endDate)) {
            this.isRented = true;
            this.rentalStartDate = startDate;
            this.rentalEndDate = endDate;
            return true;
        }
        return false;
    }

    public String getRentalDetails() {
        if (isRented) {
            return "Rented from " + rentalStartDate + " to " + rentalEndDate;
        }
        return "Available";
    }
}
